tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/Analyzer.scala

```diff
@@ -349,7 +350,80 @@ final class Analyzer(semantics: Semantics, outputMode: OutputMode,
           None
         }
       }
-      loop(this)
+      val existing =
+        if (isScalaClass) loop(this)
+        else methodInfos.get(methodName).filter(!_.isAbstract)
+
+      if (!allowAddingSyntheticMethods) {
+        existing
+      } else if (existing.exists(m => !m.isDefaultBridge || m.owner == this)) {
+        /* If we found a non-bridge, it must be the right target.
+         * If we found a bridge directly in this class/interface, it must also
+         * be the right target.
+         */
+        existing
+      } else {
+        // Try and find the target of a possible default bridge
+        findDefaultTarget(methodName).fold {
+          existing
+        } { defaultTarget =>
+          if (existing.exists(_.defaultBridgeTarget == defaultTarget.owner.encodedName)) {
+            /* If we found an existing bridge targeting the right method, we
+             * can reuse it.
+             * We also get here with None when there is no target whatsoever.
+             */
+            existing
+          } else {
+            // Otherwise, create a new default bridge
+            Some(createDefaultBridge(defaultTarget))
+          }
+        }
+      }
+    }
+
+    /** Resolves an inherited default method.
+     *
+     *  This lookup is specified by the JVM resolution rules for default
+     *  methods. See the `invokespecial` opcode in the JVM Specification
+     *  version 8, Section 6.5:
+     *  https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-6.html#jvms-6.5.invokespecial
+     */
+    private def findDefaultTarget(methodName: String): Option[MethodInfo] = {
+      val candidates = for {
+        intf <- ancestors if intf.isInterface
+        m <- intf.methodInfos.get(methodName)
+        if !m.isAbstract && !m.isDefaultBridge
+      } yield m
+
+      val notShadowed = candidates filterNot { m =>
+        candidates exists { n =>
+          (n ne m) && n.owner.ancestors.contains(m.owner)
+        }
+      }
+
+      /* Deviation from the spec: if there are several targets, the spec
+       * chooses one arbitrarily. We fail the linking instead, so that we can
+       * more easily discover errors. Note that the Java *language* and
+       * compiler do not let this happen (in the absence of manipulated
+       * classpaths, of course).
+       */
+      if (notShadowed.size == 1) Some(notShadowed.head)
+      else None
```

Comment:
Ah but there's no `From` available at this point ... Which makes sense, since lookups were never supposed to error out. Even non-try lookups do not fail. They create non-existent methods, and it is only when they are reached that they produce errors.

So it should create a conflicting default bridge (similar to a non-existent method) which, when reached, would produce an error. But even that is hard, because try-lookups should not create non-existent stuff, only `lookupMethod` is supposed to produce erroneous methods. I guess `tryLookupMethod` should then return a `List[MethodInfo]` rather than an `Option[MethodInfo]`, and it is then the job of `lookupMethod` to produce an erroneous method if it receives more than one target. But that is likely going to slow down the analyzer as a whole, which is already a bottleneck of the pipeline.

I don't like this ... I'm tempted to just follow the JVM spec and return `notShadowed.headOption` without error. Or possibly send a warning to `Console.err` (we don't even have a logger here).